packages/devtools-source-map/src/source-map.js

```diff
@@ -367,9 +367,13 @@ async function getGeneratedRangesForOriginal(
     map.computeColumnSpans();
   }
 
-  const cachedGeneratedMappingsForOriginal = GENERATED_MAPPINGS.get(map);
-  if (cachedGeneratedMappingsForOriginal) {
-    return cachedGeneratedMappingsForOriginal;
+  if (!GENERATED_MAPPINGS.has(map)) {
+    GENERATED_MAPPINGS.set(map, new Map());
+  }
+
+  const generatedRangesMap = GENERATED_MAPPINGS.get(map);
+  if (generatedRangesMap.has(sourceId)) {
```

Comment:
Ugh, thanks! Could we maybe cache this based on `url` instead, since that's the value we actually use for calculating the data?